storage/types.go

```diff
@@ -370,6 +370,32 @@ func (n *NodeID) MaskLeft(depth int) *NodeID {
 	}
 }
 
+// SetsRight all the bits to the right of depth to val.
+func (n *NodeID) SetRight(depth int, val byte) *NodeID {
+	r := make([]byte, len(n.Path))
+	// Set the right bytes to val, excluding the overlap byte
+	depthBytes := bytesForBits(depth)
+	vals := bytes.Repeat([]byte{val}, len(n.Path)-depthBytes)
+	copy(r[depthBytes:], vals)
+
+	if depth > 0 {
+		// Copy the first depthBytes-1.
+		copy(r, n.Path[:depthBytes-1])
+		// Set the first leftmask bits of the last byte.
+		left := n.Path[depthBytes-1] & leftmask[depth%8]
+		right := val &^ leftmask[depth%8]
+		r[depthBytes-1] = left | right
+	}
+	b := n.PrefixLenBits
+	if depth < n.PrefixLenBits {
+		b = depth
```

Comment:
Why do you change the bit length here? Isn't this method supposed to only flip bits?

Reply:
Good point. I'll remove it.